src/main.rs

```diff
@@ -32,38 +32,21 @@ mod bootstrap;
 mod cmd;
 mod control;
 mod data;
+mod shell;
 mod transfer;
 
 /* =================================================
-   main() function
+   helper functions
    =================================================
 */
-#[tokio::main]
-async fn main() -> Result<(), Box<dyn Error>> {
-    env_logger::init();
-
-    let c = cmd::initialize_from_args();
-    debug!("Command details: {:?}", c);
-
-    // when --region <region-name e.g. ap-northeast-1>, use the region. when --region local, use DynamoDB local.
-    // --region/--table option can be passed as a top-level or subcommand-level (i.e. global).
-    let mut context = app::Context {
-        region: None,
-        config: Some(app::load_or_touch_config_file(true)?),
-        cache: Some(app::load_or_touch_cache_file(true)?),
-        overwritten_region: app::region_from_str(c.region),
-        overwritten_table_name: c.table,
-        output: None,
-    };
-    debug!("Initial command context: {:?}", &context);
-
-    match c.child {
+async fn dispatch(context: &mut app::Context, subcommand: cmd::Sub) -> Result<(), Box<dyn Error>> {
```

Comment:
Changed `context` from `app::Context` to `&mut app::Context` and put `clone()` around.
Eliminating `clone()`s need further refactoring thus I won't do in this P-R.
I guess it is related to #6 and should be done in that fix.

Reply:
Yes, in #6 we'll deal with the "too much `clone()`s" issue. So it's ok to add `clone()`s to make the feature work in this PR.